app/src/main/java/com/google/android/sambadocumentsprovider/browsing/BrowsingAdapter.java

```diff
@@ -0,0 +1,268 @@
+package com.google.android.sambadocumentsprovider.browsing;
```

Comment:
This kinda feels more related to UI rather than browsing logic. Put it under .mount where MountActivity resides.